src/proof_system/proof.rs

```diff
@@ -54,7 +54,7 @@ pub struct Proof {
 
 impl Proof {
     /// Performs the verification of a `Proof` returning a boolean result.
-    pub fn verify(
+    pub(crate) fn verify(
```

Comment:
It is done like that because the verifier is expected to use the `Verify` struct, yes?

Reply:
It's done like this because the `Verifier` struct is indeed using this function. But you would never be able to use it since you don't have access to the fields required by the fn signature since they're internal to the `Verifier` struct.

Therefore, the function is needed, but should not be public since it can't be called/used by the end user.